sdk/storage/azure-storage-blob/azure/storage/blob/__init__.py

```diff
@@ -81,3 +85,73 @@
     'ResourceTypes',
     'AccountPermissions',
 ]
+
+
+def upload_blob_to_url(
+        blob_url,  # type: str
+        data,  # type: Union[Iterable[AnyStr], IO[AnyStr]]
+        overwrite=False,  # type: bool
+        max_connections=1,  # type: int
+        encoding='UTF-8', # type: str
+        credential=None,  # type: Any
+        **kwargs):
+    """Upload data to a given URL
+
+    The data will be uploaded as a block blob.
+
+    :param str blob_url:
+        The full URI to the blob. This can also include a SAS token.
+    :param data:
+        The data to upload. This can be bytes, text, an iterable or a file-like object.
+    :type data: bytes or str or Iterable
+    :param bool overwrite:
+        Whether the blob to be uploaded should overwrite the current data.
+        If True, upload_blob_to_url will silently overwrite the existing data. If set to False, the
+        operation will fail with a ResourceExistsError.
+    :param credential:
+        The credentials with which to authenticate. This is optional if the
+        blob URL already has a SAS token. The value can be a SAS token string, an account
+        shared access key, or an instance of a TokenCredentials class from azure.identity.
```

Comment:
Which one wins if you provide both creds in the URL and as a credential parameter?